### PR TITLE
make vat and manager immutable

### PR DESCRIPTION
**Pull request by gbalabasquer**

### CODE REVIEW

**Comment by gbalabasquer** [comment]
@kmbarry1 @talbaneth I've assigned you guys as reviewers as lately were checking the other proxy actions. This change is in line with what was done in charter proxy actions.
I've left all the deprecated functions as existing ones just in case but I'm seriously thinking to remove them, first as I'm not even sure the contract can be deployed this big but secondly to avoid so much old functions in the code that might bring confusion to readers.
To be honest there is not need to migrate to a new proxy actions until you are ready, so integrators could still use the old ones even forever if they want. But let me know if you have any other thought, otherwise I'll probably proceed to remove them.

File: src/DssProxyActions.sol [hunk 1]
```diff
@@ -888,22 +1196,34 @@ contract DssProxyActionsEnd is Common {
...
+    // Deprecated
     function freeETH(
-        address manager,
+        address manager_,
         address ethJoin,
         address end,
         uint256 cdp
     ) public {
-        uint256 wad = _free(manager, end, cdp);
+        require(manager_ == address(manager), "DssProxyActions/Non-matching-manager");
```
Comment (talbaneth) [inline src/DssProxyActions.sol:1216]: Should probaby update the revert messages contract to DssProxyActionsEnd.
Same in similar places below.

File: src/DssProxyActions.sol [hunk 2]
```diff
@@ -427,38 +572,49 @@ contract DssProxyActions is Common {
...
     function safeLockGem(
-        address manager,
         address gemJoin,
         uint256 cdp,
         uint256 amt,
         address owner
     ) public {
         require(ManagerLike(manager).owns(cdp) == owner, "owner-missmatch");
-        lockGem(manager, gemJoin, cdp, amt);
+        lockGem(gemJoin, cdp, amt);
```
Comment (talbaneth) [inline src/DssProxyActions.sol:594]: No need for ManagerLike casting on line 593

**Review by talbaneth — COMMENTED** [review]
Hi,
See small comments.
One small difference between this and what I did in charter's proxy actions was that I stored vat and manager as addresses and not VatLike and ManagerLike.
Done that mainly because I saw it was the convention in most make contracts, but I do agree it's fine either way.

In gerneral I also think we should remove the deprecated functions, as indeed anyone can still use the old proxy actions or create one as he fits.

File: src/DssProxyActions.sol [hunk 3]
```diff
@@ -813,71 +1095,97 @@ contract DssProxyActions is Common {
...
         // Moves the amount from the CDP urn to proxy's address
-        flux(manager, cdp, address(this), wadC);
+        flux(cdp, address(this), wadC);
         // Exits token amount to the user's wallet as a token
         GemJoinLike(gemJoin).exit(msg.sender, amtC);
     }
 }
 
 contract DssProxyActionsEnd is Common {
+    ManagerLike immutable manager;
```
[the remark this answers is not in the document]
Comment (gbalabasquer) [reply src/DssProxyActions.sol:770]: Fixed

File: src/DssProxyActions.sol [hunk 4]
```diff
@@ -499,42 +677,62 @@ contract DssProxyActions is Common {
...
-                ManagerLike(manager).urns(cdp),
-                ManagerLike(manager).ilks(cdp),
+                manager.urns(cdp),
+                manager.ilks(cdp),
                 wad
             )
         );
         // Moves the DAI amount (balance in the vat in rad) to proxy's address
-        move(manager, cdp, address(this), _toRad(wad));
+        move(cdp, address(this), _toRad(wad));
```
[the remark this answers is not in the document]
Comment (gbalabasquer) [reply src/DssProxyActions.sol:492]: Changed.

File: src/DssProxyActions.sol (hunk 1 above)
Comment (gbalabasquer) [reply src/DssProxyActions.sol]: Not done as I've removed the deprecated functions.

File: src/DssProxyActions.sol (hunk 2 above)
Comment (gbalabasquer) [reply src/DssProxyActions.sol:412]: Done.

File: src/DssProxyActions.sol [hunk 5]
```diff
@@ -110,6 +110,12 @@ interface ProxyLike {
 contract Common {
     uint256 constant RAY = 10 ** 27;
 
+    VatLike     immutable vat;
```
Comment (talbaneth) [inline src/DssProxyActions.sol:113]:
```suggestion
    VatLike     immutable public vat;
```

File: src/DssProxyActions.sol [hunk 6]
```diff
@@ -130,6 +136,12 @@ contract Common {
 }
 
 contract DssProxyActions is Common {
+    ManagerLike immutable manager;
```
Comment (talbaneth) [inline src/DssProxyActions.sol:139]:
```suggestion
    ManagerLike immutable public manager;
```

File: src/DssProxyActions.sol (hunk 3 above)
[the remark this answers is not in the document]
Comment (talbaneth) [reply src/DssProxyActions.sol:770]: github won't let me suggest a fix on this one, but it should be public as well.

**Review by kmbarry1 — APPROVED** [review]
Don't see any issues here, although making the values public like Tal suggests may be convenient.

File: src/DssProxyActions.sol (hunk 3 above)
[the remark this answers is not in the document]
Comment (gbalabasquer) [reply src/DssProxyActions.sol]: Done.

File: src/DssProxyActions.sol (hunk 5 above)
Comment (gbalabasquer) [reply src/DssProxyActions.sol]: Done.

File: src/DssProxyActions.sol (hunk 6 above)
Comment (gbalabasquer) [reply src/DssProxyActions.sol]: Done.

**Comment by gbalabasquer** [comment]
@talbaneth just pushed the last suggestions, please approve if everything is ok and I'll get this merged.
Thanks!

**Review by talbaneth — APPROVED** [review]
LGTM

**Pull request merged**